src/descriptor/satisfied_constraints.rs

```diff
@@ -257,24 +260,14 @@ impl<'desc, 'stack, F> SatisfiedConstraints<'desc, 'stack, F>
 where
     F: FnMut(&bitcoin::PublicKey, BitcoinSig) -> bool,
 {
-    /// Helper function to push a NodeEvaluationState on state stack
-    fn push_evaluation_state(
-        &mut self,
-        node: &'desc Miniscript<bitcoin::PublicKey>,
-        n_evaluated: usize,
-        n_satisfied: usize,
-    ) -> () {
-        self.state.push(NodeEvaluationState {
-            node,
-            n_evaluated,
-            n_satisfied,
-        })
-    }
-
-    /// Creates a new iterator over all constraints satisfied for a given
+    // Creates a new iterator over all constraints satisfied for a given
     /// descriptor by a given witness stack. Because this iterator is lazy,
     /// it may return satisfied constraints even if these turn out to be
     /// irrelevant to the final (dis)satisfaction of the descriptor.
+    /// Because the iterator is dependent evaluating the miniscript, it is
+    /// dependent on scriptContext. Since, the caller of the function should
+    /// not care about Miniscript Context, resulting contextis transformed into
+    /// ScriptContext::Any
```

Comment:
We can remove this extra comment now that it's an invisible implementation detail.